lib/instrumentation/generic-span.js

```diff
@@ -101,3 +109,9 @@ GenericSpan.prototype.addLabels = function (labels) {
   }
   return true
 }
+
+GenericSpan.prototype.setType = function (type, subtype, action) {
+  this.type = type || 'custom'
```

Comment:
I think if we set the initial value to `custom` (see constructor), then we can just drop the fallback here right?

```suggestion
  this.type = type
```

Reply:
yeah, I did consider that but reluctantly concluded that this is already possible with the current implementation. So while it's not ideal, we're not introducing a "bug". I think the real solution is to move the `custom` fallback to the encoding step at which time there's no way for the user to later overwrite it with an invalid value. What do you think of that?

Reply:
I moved it to the encoding functions.